activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -746,10 +747,20 @@ def pk_and_sequence_for(table)
         end
       end
 
-      # Returns just a table's primary key
-      def primary_key(table)
-        pk_and_sequence = pk_and_sequence_for(table)
-        pk_and_sequence && pk_and_sequence.first
+      def primary_keys(table_name) # :nodoc:
+        return [] unless table_name.present?
```

Comment:
Prefer to raise an `ArgumentError` for an exceptional case like this. What is passing `nil` or blank table names?